internal/tagencoding/tagencoding.go

```diff
@@ -38,14 +32,11 @@ func (vb *Values) growIfRequired(expected int) {
 }
 
 func (vb *Values) WriteValue(v []byte) {
-	length := len(v)
-	vb.growIfRequired(sizeOfUint16 + length)
+	length := len(v) & 0xff
```

Comment:
length := byte(len(v))

Reply:
That's not clearly any better to me -- length is used as an int on lines 36 and 49, so your way is 3 type conversions instead of 1.  :)